starport_template/lib/pages/assets_portfolio_page.dart

```diff
@@ -0,0 +1,181 @@
+import 'package:cosmos_ui_components/components/content_state_switcher.dart';
+import 'package:cosmos_ui_components/components/cosmos_elevated_button.dart';
+import 'package:cosmos_ui_components/components/gradient_avatar.dart';
+import 'package:cosmos_ui_components/components/template/cosmos_balance_card.dart';
+import 'package:cosmos_ui_components/components/template/cosmos_wallets_list_view.dart';
+import 'package:cosmos_ui_components/cosmos_theme.dart';
+import 'package:cosmos_utils/amount_formatter.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_mobx/flutter_mobx.dart';
+import 'package:mobx/mobx.dart';
+import 'package:starport_template/entities/balance.dart';
+import 'package:starport_template/starport_app.dart';
+
+class AssetsPortfolioPage extends StatefulWidget {
+  final WalletInfo walletInfo;
+
+  const AssetsPortfolioPage({Key? key, required this.walletInfo}) : super(key: key);
+
+  @override
+  _AssetsPortfolioPageState createState() => _AssetsPortfolioPageState();
+}
+
+class _AssetsPortfolioPageState extends State<AssetsPortfolioPage> {
+  ObservableList<Balance> get balancesList => StarportApp.walletsStore.balancesList;
+
+  bool get isBalancesLoading => StarportApp.walletsStore.isBalancesLoading;
+
+  bool get isSendMoneyLoading => StarportApp.walletsStore.isSendMoneyLoading;
+
+  bool get isError => StarportApp.walletsStore.isBalancesLoadingError;
+
+  @override
+  void initState() {
+    super.initState();
+    _fetchWalletBalances();
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
the widget structure is a bit deep and complex, could we extract some to separate Widgets/ builder methods for readability?

Reply:
Updated